service/history/historyEngine.go

```diff
@@ -2173,6 +2175,23 @@ func (e *historyEngineImpl) RemoveSignalMutableState(ctx context.Context, reques
 		})
 }
 
+func (e *historyEngineImpl) validateTerminateRequest(ctx context.Context, domainID string, execution workflow.WorkflowExecution) (retError error) {
+	context, release, err0 := e.historyCache.getOrCreateWorkflowExecutionWithTimeout(ctx, domainID, execution)
+	if err0 != nil {
+		return err0
+	}
+	defer func() { release(retError) }()
+	msBuilder, retError := context.loadWorkflowExecution()
+	if retError != nil {
+		return retError
+	}
+	// terminating a WF that just get started would be wasting resources, and causing problem without eventsV2
+	if time.Now().Before(msBuilder.GetExecutionInfo().StartTimestamp.Add(time.Second * 3)) {
```

Comment:
make this a constant or a dynamic config

Reply:
one more thing, this patch is "treating the symptom not the disease", the correct way is to include the previous run id in the start event, so when the start event is applied, order between run ID can be maintained.

Reply:
This will also allow customer to start-terminate-start within very short time of period which is wasting resources. Or another way I am thinking is to reject starting the same wfID within short time.